File: r_barcode/android/src/main/kotlin/com/rhyme/r_barcode/RBarcodeEventChannel.kt
package com.rhyme.r_barcode

import android.os.Handler
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.EventChannel.EventSink

class RBarcodeEventChannel(messenger: BinaryMessenger?, eventChannelId: Long) : EventChannel.StreamHandler {
    private var eventSink: EventSink? = null
    private val scanViewType = "com.rhyme_lph/r_barcode"
    private val handle: Handler = Handler()

    init {
        EventChannel(messenger, scanViewType + "_" + eventChannelId + "/event").setStreamHandler(this)
    }

    fun sendMessage(msg: Any) {
        handle.post {
            eventSink?.success(msg)
        }
    }

    fun dispose() {
        eventSink = null
    }

    override fun onListen(arguments: Any?, sink: EventSink?) {
        eventSink = sink

    }

    override fun onCancel(arguments: Any?) {
        eventSink = null

    }
}

File: r_barcode/ios/Classes/RBarcodeEventChannel.swift
//
//  RBarcodeEventChannel.swift
//  r_barcode
//
//  Created by 李鹏辉 on 2020/6/7.
//  rhymelph@gmail.com
//

import Foundation


class RBarcodeEventChannel: NSObject , FlutterStreamHandler {
    var events: FlutterEventSink?
    
    init(textureId:Int64,messenger:FlutterBinaryMessenger) {
        super.init()
        let eventChannel = FlutterEventChannel(name: "com.rhyme_lph/r_barcode_\(textureId)/event", binaryMessenger: messenger)
        eventChannel.setStreamHandler(self)
    }
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.events = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        events = nil
        return nil
    }
    
    func sendMessage(msg:Any?) {
        events?(msg)
    }
    
}
